assets/javascripts/modules/toggleDynamicFormFields.js

```diff
@@ -1,33 +1,39 @@
 require('jquery');
 
-module.exports = function() {
+module.exports = function () {
   var $dynamicForm = $('form[data-dynamic-form]');
-
-  $dynamicForm.find('*[data-dynamic-fields-hide="true"]').each(function() {
-    if ($(this).prop('checked')) {
-      var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
-      $toggledField.addClass('js-hidden');
-    }
-  });
-
-  $dynamicForm.on('click', '*[data-dynamic-fields]', function() {
-    var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
-    if ($(this).data('dynamic-fields-hide')) {
-      if ($(this).attr('type') === 'checkbox') {
-        $(this).data('dynamic-fields-hide', false);
+   $dynamicForm.find('*[data-dynamic-fields-hide="true"]').each(function () {
+      if ($(this).prop('checked')) {
+          var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
+          $toggledField.addClass('js-hidden');
       }
-
-      $toggledField.addClass('js-hidden');
-      $toggledField.find(':text').val('');
-      $toggledField.find(':checked').prop('checked', false);
-      $toggledField.find('*[data-default]').prop('checked', true);
-    } else {
-      if ($(this).attr('type') === 'checkbox') {
-        $(this).data('dynamic-fields-hide', true);
+      $('.data-' + $(this).data('nested-children')).each(function () {
+          if (!$(this).children('div').hasClass("js-hidden")) {
+              $(this).children('label').children('input[type=checkbox]').data('dynamic-fields-hide', true);
+          }
+      });
+  });
```

Comment:
After discussing more it seems there may have to be a change to the logic.

Here is some pseudo code to get my approach across:
This code would replace lines 5-15
```
 $dynamicForm.find('[data-dynamic-fields-hide]').each(function (index, dynamicField) {
    var $dynamicFieldTriggerElement = $(dynamicField);
    var $dynamicFieldElement = $('.data-' + $dynamicFieldTriggerElement.prop('dynamic-fields'));
    var hideDynamicField = $dynamicFieldTriggerElement.prop('data-dynamic-fields-hide');
    
    if (hideDynamicField) {
      if ($(this).prop('checked')) {
        $dynamicFieldElement.addClass('js-hidden');
      }
    } else {
      if ($(this).prop('checked')) {
        $dynamicFieldElement.removeClass('js-hidden');
      }
    }
  });
```